hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -878,6 +879,24 @@ private synchronized void addResourceObject(Resource resource) {
   private static final int SUB_START_IDX = 0;
   private static final int SUB_END_IDX = SUB_START_IDX + 1;
 
+  private Set<String> PROXY_USERS = null;
+
+  public synchronized Set<String> getProxyUsers() {
```

Comment:
Performance wise maybe it's better not to synchronize the method. This pattern might be better https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html